src/class-post-type.php

```diff
@@ -0,0 +1,85 @@
+	 * @return array                      Labels
+	 */
+	public static function generate_labels( string $singular, string $plural, string $translation_domain ) {
+		$singular_lowercase = strtolower( $singular );
+		$plural_lowercase   = strtolower( $plural );
+
+		// phpcs:disable WordPress.WP.I18n
+		$labels = [
+			'name'                  => __( $plural, $translation_domain ),
+			'singular_name'         => __( $singular, $translation_domain ),
+			'menu_name'             => __( $plural, $translation_domain ),
+			'name_admin_bar'        => __( $singular, $translation_domain ),
+			'all_items'             => __( 'All ' . $plural_lowercase, $translation_domain ),
+			'add_new_item'          => __( 'Add a ' . $singular_lowercase, $translation_domain ),
+			'edit_item'             => __( 'Edit ' . $singular_lowercase, $translation_domain ),
+			'new_item'              => __( 'New ' . $singular_lowercase, $translation_domain ),
+			'view_item'             => __( 'See ' . $singular_lowercase, $translation_domain ),
+			'view_items'            => __( 'See ' . $plural_lowercase, $translation_domain ),
+			'archives'              => __( $plural_lowercase . ' archives', $translation_domain ),
+			'attributes'            => __( $plural_lowercase . ' attributs', $translation_domain ),
+			'filter_items_list'     => __( 'Filter ' . $plural_lowercase . ' list', $translation_domain ),
+			'items_list'            => __( $plural_lowercase . ' list', $translation_domain ),
+			'items_list_navigation' => __( $plural_lowercase . ' list navigation', $translation_domain ),
+			'search_items'          => __( 'Search ' . $plural_lowercase, $translation_domain ),
+			'not_found'             => __( 'No ' . $singular_lowercase, $translation_domain ),
+			'not_found_in_trash'    => __( 'No ' . $singular_lowercase . ' in trash', $translation_domain ),
+		];
+		// phpcs:enable
+
+		return $labels;
+	}
+
+	/**
+	 * Create a custom post type
+	 *
+	 * @param  string       $singular           Singular name.
+	 * @param  string       $plural             Plural name.
+	 * @param  string|null  $translation_domain Translation domain name.
+	 * @param  array        $args               Custom arguments.
+	 * @param  bool|boolean $public             Is post type public?.
+	 * @param  array        $supports           Supported features.
+	 * @return array                            Custom post type
+	 */
+	public static function create( string $singular, string $plural, string $translation_domain = null, array $args = array(), bool $public = true, array $supports = array() ) {
+		$singular_lowercase = strtolower( $singular );
+
+		$labels = self::generate_labels( $singular, $plural, $translation_domain );
+
+		// phpcs:disable WordPress.WP.I18n
+		$default_args = array(
+			'label'              => __( $singular, $translation_domain ),
+			'labels'             => $labels,
+			'public'             => true,
+			'publicly_queryable' => $public,
```

Comment:
This is strange to use a parameter `$public` to alter the default args, and another to override the default args `$args` with array_merge. Maybe `$public` is not needed ?

Reply:
I've do this because the `public` argument is an important argument (generates public URLs).
In my opinion, it requires a specific parameter to don't miss it.

Reply:
it is maybe missleading to have `'publicly_queryable' => $public`, should we use the same variable name ? 

I also think that setting publicly_queryable to false could lead to issues because you will not be able to use `WP_Query( array( 'post_type' => 'cpt' );` i think this will silently fail to give results. Could you test this behavior ?

Reply:
I'm agree to change variable `$public` name to a more specific name like `publicly_queryable`.

The `publicly_queryable` arg manage the visibility of the post type on the front-end. When set to `false` he's still queryable through WP_Query, tested and already used on some projects.